From f57c318e23e7d39f4108a2a9844ec9126096fae0 Mon Sep 17 00:00:00 2001
From: Long Dang <long.dang.aj@rvc.renesas.com>
Date: Fri, 12 Aug 2016 13:31:29 +0700
Subject: [PATCH 40/42] gst145-omxvideodec: Avoid detaching buffer metadata
 when returning to the buffer pool

Porting from a patch in Linux 3.10 environment made by Kazunori Kobayashi <kkobayas@igel.co.jp>

The buffer pool removes all metadata without the GST_META_FLAG_POOLED
flag in the buffre release processing. This change suppresses that.

Signed-off-by: Long Dang <long.dang.aj@rvc.renesas.com>
---
 omx/gstomxbufferpool.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/omx/gstomxbufferpool.c b/omx/gstomxbufferpool.c
index ddc28a8..52128b6 100644
--- a/omx/gstomxbufferpool.c
+++ b/omx/gstomxbufferpool.c
@@ -676,6 +676,8 @@ gst_omx_buffer_pool_acquire_buffer (GstBufferPool * bpool,
           new_buf = gst_omx_buffer_pool_request_videosink_buffer_creation (pool,
               dmabuf_fd, vmeta->stride);
         else {
+          GstVideoMeta *new_meta;
+
           new_buf = gst_buffer_new ();
           for (i = 0; i < n_planes; i++)
             gst_buffer_append_memory (new_buf,
@@ -688,6 +690,11 @@ gst_omx_buffer_pool_acquire_buffer (GstBufferPool * bpool,
               GST_VIDEO_INFO_HEIGHT (&pool->video_info),
               GST_VIDEO_INFO_N_PLANES (&pool->video_info), vmeta->offset,
               vmeta->stride);
+
+          new_meta = gst_buffer_get_video_meta (new_buf);
+          /* To avoid detaching meta data when a buffer returns
+             to the buffer pool */
+          GST_META_FLAG_SET (new_meta, GST_META_FLAG_POOLED);
         }
   
         g_ptr_array_remove_index (pool->buffers, pool->current_buffer_index);
-- 
1.9.1

